src/pages/AccountDetails.vue

```diff
@@ -19,55 +16,93 @@ import InstructionsTable from '@/shared/ui/components/InstructionsTable.vue';
 
 const router = useRouter();
-const { handleUnknownError } = useErrorHandlers();
+const { width } = useWindowSize();
+
+const hashType = computed(() => {
+  if (width.value > XS_WINDOW_SIZE) return 'medium';
+
+  return 'short';
+});
 
 const accountId = computed(() => {
   const id = router.currentRoute.value.params['id'];
 
   return AccountIdSchema.parse(id);
 });
 
-const account = ref<Account | null>(null);
-const isFetchingAccount = ref(false);
-
-onMounted(async () => {
-  try {
-    isFetchingAccount.value = true;
-    account.value = await http.fetchAccount(accountId.value);
+const accountScope = useParamScope(
+  () => {
+    return {
+      key: accountId.value.toString(),
+      payload: accountId.value,
+    };
+  },
+  ({ payload }) => setupAsyncData(() => http.fetchAccount(payload))
+);
 
-    if (account.value) {
-      const ownedEntities = [];
+const isAccountLoading = computed(() => accountScope.value.expose.isLoading);
+const account = computed(() => {
+  const res = accountScope.value?.expose.data;
 
-      if (account.value.owned_assets) ownedEntities.push(assetsTable.fetch({ owned_by: accountId.value.toString() }));
-      if (account.value.owned_domains) ownedEntities.push(domainsTable.fetch({ owned_by: accountId.value.toString() }));
+  if (!res) return null;
 
-      if (!ownedEntities.length) return;
+  return res;
+});
 
-      await Promise.all(ownedEntities);
-    }
-  } catch (e) {
-    handleUnknownError(e);
-  } finally {
-    isFetchingAccount.value = false;
-  }
+const domainsListState = reactive({
+  page: 1,
+  per_page: 10,
+  owned_by: computed(() => accountId.value),
 });
 
-const assetsTable = useTable(http.fetchAssets);
-const domainsTable = useTable(http.fetchDomains);
+watch([() => domainsListState.per_page], () => {
+  domainsListState.page = 1;
+});
 
-const transactionsTab = ref<TabAccountTransactions>('transactions');
+const domainsScope = useParamScope(
+  () => {
+    return {
+      key: account.value?.owned_domains ? JSON.stringify(domainsListState) : '',
+      payload: domainsListState,
+    };
+  },
+  ({ payload }) => setupAsyncData(() => http.fetchDomains(payload))
+);
+
+const isDomainsLoading = computed(() => domainsScope.value?.expose?.isLoading);
+const totalDomains = computed(() => domainsScope.value?.expose?.data?.pagination?.total_items ?? 0);
+const domains = computed(() => domainsScope.value?.expose?.data?.items ?? []);
+
+const assetsListState = reactive({
+  page: 1,
+  per_page: 10,
+  owned_by: computed(() => accountId.value),
+});
 
-const shouldShowInstructions = computed(() => transactionsTab.value === 'instructions');
+watch([() => assetsListState.per_page], () => {
+  assetsListState.page = 1;
+});
 
-const { width } = useWindowSize();
+const assetsScope = useParamScope(
+  () => {
+    return {
+      key: account.value?.owned_assets ? JSON.stringify(assetsListState) : '',
+      payload: assetsListState,
+    };
+  },
```

Comment:
Please read the documentation of `useParamScope`. It supports optional scopes:
```suggestion
  () => {
    return (account.value?.owned_assets || false) && {
      key: JSON.stringify(assetsListState),
      payload: assetsListState,
    };
  },
```

Reply:
Great! I decided to use `if` condition instead and return either `null` or scope.